README.md

```diff
@@ -48,7 +48,8 @@ RDM is designed to be used within a typical software development workflow.  When
 8. Pull requests must be reviewed, and certain standardized comments are placed in reviews to confirm validation
 9. Write new architecture documents as new _software items_ are implemented
 10. Once a new _release_ is cut, generate a set of IEC62304 documents using `rdm release`
-11. These markdown files can then be converted to PDFs or Word documents using a tool such as [Pandoc](https://pandoc.org)
+11. Run `rdm gap checklists/[some checklist] release/*.md` to check for missing items
```

Comment:
Two questions about this line:

1. Is it possible to run `rdm gap ./mychecklist.txt file1.md file2.md`?
2. For the built-in checklists, what is the format? E.g., is it `rdm gap 62304_base_class_c.txt`?

Reply:
1. Yes it is possible. I was trying to show how to do it with the documents generated for release. On that topic, how do we handle the AuditNoteExclusionExtension vs. AuditNoteInclusionExtension choice for auditor notes? The exclude option will fail the gap audit almost completely.

2. There is currently no logic to automatically find the packaged (i.e. "built-in") checklists. A full path is needed to where ever it is.

Reply:
2. There is currently no logic to automatically find the packaged (i.e. "built-in") checklists. A full path is needed to where ever it is.

I think we will need this, since users won't generally know/care where the checklists are stored (this is also why some sort of listing of the builtin lists would be a useful commandline option).

Reply:
> On that topic, how do we handle the AuditNoteExclusionExtension vs. AuditNoteInclusionExtension choice for auditor notes? The exclude option will fail the gap audit almost completely.

IMO we don't need to worry about the gap analysis failing if audit notes are off.

Reply:
The tool now has a checklists command (`rdm checklists`) that lists all of the built-in checklists. It also searches those built-ins when doing gap analysis and a checklist is not found in the usual directory.

Reply:
@orwonthe what was the reasoning behind adding a new command (i.e., `rdm checklists`) as opposed to adding an argument to the gap command (i.e., `rdm gap --checklists` or `rdm gap --list`)?

Reply:
> It also searches those built-ins when doing gap analysis and a checklist is not found in the usual directory.

What do you think about this idea:

If you run `rdm gap CHECKLIST FILES` and `CHECKLIST` is an absolute or relative path, look for a file, but if `CHECKLIST` is just a word (e.g., "62304_2007_class_a") then use the builtin?

This approach would work the same way bash does when it looks up an executable to run.

I like this better then having a fall-through, because it is more explicit and because you won't be able to "shadow" a checklist by the presence of a file.

For example, if you have your own custom checklist that you call `62304_2007_class_a.txt` and from within that checklist it has a line that says `include 62304_2007_class_a.txt`, will it get a circular import warning? or does it have a special case to know not to import itself?

If we require that `CHECKLIST` be a relative path for the command to use a local file, then there won't be any ambiguity or special cases. Thus, `rdm gap ./62304_2007_class_a.txt` (using the local file) would be different than `rdm gap 62304_2007_class_a.txt` (using the builtin). It may seem confusing, but since this is how bash works, perhaps the distinction will feel natural to people.

Reply:
> @orwonthe what was the reasoning behind adding a new command (i.e., `rdm checklists`) as opposed to adding an argument to the gap command (i.e., `rdm gap --checklists` or `rdm gap --list`)?

The command line syntax was easier since neither a checklist or source files are needed where as the gap command needs both. Making it a separate command is simpler to explain to both the user and python.

Reply:
> Making it a separate command is simpler to explain to both the user and python.

Its not clear to me whether this is simpler for the user.

Will the checklists ever be used outside of the `rdm gap` command? I don't think so... thus it feels natural to me to have it under the same command.

E.g., it would be as if `git branch --list` was its own command, say `git branch-list`.

Reply:
> What do you think about this idea:
> 
> If you run `rdm gap CHECKLIST FILES` and `CHECKLIST` is an absolute or relative path, look for a file, but if `CHECKLIST` is just a word (e.g., "62304_2007_class_a") then use the builtin?

I like the idea. I actually began implementing it but then decided against it. But I think the implementation I chose is going to be prone to obscure bugs when people get the folders and paths confused. I am going to redo it as you suggest.